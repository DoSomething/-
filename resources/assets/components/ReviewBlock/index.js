import React from 'react';
import PropTypes from 'prop-types';

import Tags from '../Tags';
import StatusButton from '../StatusButton';

class ReviewBlock extends React.Component {
  setStatus(status) {
    return this.props.onUpdate(this.props.post, { status });
  }

  render() {
    const post = this.props.post;
    const disableTags = !(
      post.status === 'accepted' || post.status === 'rejected'
    );

    return (
      <div>
        <ul className="form-actions -inline" style={{ marginTop: 0 }}>
          <li>
            <StatusButton
              type="accepted"
              label="accept"
              status={post.status}
              setStatus={this.setStatus.bind(this)}
            />
          </li>
          <li>
            <StatusButton
              type="rejected"
              label="reject"
              status={post.status}
              setStatus={this.setStatus.bind(this)}
            />
          </li>
        </ul>
        <ul className="form-actions -inline">
          <li>
            <button
              className="button delete -tertiary"
              onClick={e => this.props.deletePost(post.id, e)}
            >
              Delete
            </button>
          </li>
        </ul>
        <Tags
          id={post.id}
          tagged={post.tags}
          onTag={this.props.onTag}
          disabled={disableTags}
        />
      </div>
    );
  }
}

ReviewBlock.propTypes = {
  deletePost: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired,
  onTag: PropTypes.func.isRequired,
  post: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default ReviewBlock;
