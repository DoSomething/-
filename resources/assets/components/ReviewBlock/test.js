import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ReviewBlock from './index';

test('it renders the review block', () => {
    const post = {
      status: 'accepted',
    };

  const component = shallow(
    <ReviewBlock post={post} />
  );

  expect(toJson(component)).toMatchSnapshot();
});
