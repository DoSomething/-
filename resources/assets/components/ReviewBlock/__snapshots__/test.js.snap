// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders accepted, rejected, and delete buttons 1`] = `
  <div>
    <ul className="form-actions -inline" style={{'marginTop': 0}}>
      <li><StatusButton type="accepted" label="accept" status=accepted setStatus={[Function]}/></li>
      <li><StatusButton type="rejected" label="reject" status=rejected setStatus={[Function]}/></li>
    </ul>
    <ul className="form-actions -inline">
      <li><button className="button delete -tertiary" onClick={[Function]}>Delete</button></li>
    </ul>
  </div>
`;

exports[`it renders the review block 1`] = `
<div>
  <ul
    className="form-actions -inline"
    style={
      Object {
        "marginTop": 0,
      }
    }
  >
    <li>
      <Component
        label="accept"
        setStatus={[Function]}
        status="accepted"
        type="accepted"
      />
    </li>
    <li>
      <Component
        label="reject"
        setStatus={[Function]}
        status="accepted"
        type="rejected"
      />
    </li>
  </ul>
  <ul
    className="form-actions -inline"
  >
    <li>
      <button
        className="button delete -tertiary"
        onClick={[Function]}
      >
        Delete
      </button>
    </li>
  </ul>
  <Tags />
</div>
`;
